### ISSUE TITLE
Support aws:PrincipalArn condition key

### ISSUE THREAD

**Issue by scriptsrc** [Netflix-Skunkworks/policyuniverse#102]
Resource policies can now use `aws:PrincipalArn` condition key instead of the `Principal` section of the statement. PolicyUniverse should understand this condition and not flag the statement as Internet Accessible.

https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_condition-keys.html

```  
    {
        "Resource": "arn:aws:s3:::blahblahblah",
        "Effect": "Allow",
        "Sid": "asdf",
        "Action": [
          "s3:DeleteBucket*",
          "s3:Get*",
          "s3:Put*"
        ],
        "Condition": {
          "StringEquals": {
            "aws:PrincipalArn": "arn:aws:iam::123456789012:role/Role"
          }
        },
        "Principal": "*"
      },
```

**Comment by tweedge** [comment]
From my understanding, this would rely on the existing ARN parser, right? I don't see an immediate need to implement a new class for representing a Principal ARN specifically.

That said it'd be nice to be able to get Principal ARNs on their own from the statement, so I'd be inclined to make a new property:
```
    @property
    def condition_principalarns(self):
        return self._condition_field("principal-arn")
```
...and then change line 305 in statement.py to be something like:
```
if entry.category == "arn" or entry.category == "principal-arn":
```

**Comment by tweedge** [comment]
Can't close this as I didn't open it :sweat_smile:

**Comment by patricksanders** [comment]
Fixed by #107
